controller/orphan_controller.go

```diff
@@ -371,6 +575,55 @@ func (oc *OrphanController) updateConditions(orphan *longhorn.Orphan) error {
 	return nil
 }
 
+// If the InstanceState condition status is true, the corresponding instance is still exist in instance manager
+func (oc *OrphanController) updateInstanceStateCondition(orphan *longhorn.Orphan, instanceType longhorn.InstanceType) (err error) {
+	var instanceState longhorn.InstanceState
+	defer func() {
+		if err != nil {
+			return
+		}
+
+		status := longhorn.ConditionStatusTrue
+		if instanceState == longhorn.InstanceStateTerminated {
+			status = longhorn.ConditionStatusFalse
+		}
+
+		orphan.Status.Conditions = types.SetCondition(orphan.Status.Conditions, longhorn.OrphanConditionTypeInstanceExist, status, string(instanceState), "")
+	}()
+
+	instanceName, instanceManager, err := oc.extractOrphanedInstanceInfo(orphan)
+	im, err := oc.ds.GetInstanceManager(instanceManager)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
```

Comment:
Shoud it be `if apierrors.IsNotFound(err) {`?

```suggestion
		if apierrors.IsNotFound(err) {
```

Reply:
Sure, it should be `apierrors.IsNotFound(err)`